apps/api/src/billing/services/usage/usage.service.spec.ts

```diff
@@ -0,0 +1,282 @@
+import type { UsageRepository } from "@src/billing/repositories/usage/usage.repository";
+import type { LeaseRepository } from "@src/deployment/repositories/lease/lease.repository";
+import { UsageService } from "./usage.service";
+
+import { AkashAddressSeeder } from "@test/seeders/akash-address.seeder";
+
+describe(UsageService.name, () => {
+  let service: UsageService;
+  let usageRepository: jest.Mocked<UsageRepository>;
+  let leaseRepository: jest.Mocked<LeaseRepository>;
+
+  beforeEach(() => {
+    usageRepository = {
+      getHistory: jest.fn()
+    } as unknown as jest.Mocked<UsageRepository>;
+
+    leaseRepository = {
+      countByDseqAndOwner: jest.fn()
+    } as unknown as jest.Mocked<LeaseRepository>;
+
+    service = new UsageService(usageRepository, leaseRepository);
+  });
+
+  describe("getHistory", () => {
+    const address = AkashAddressSeeder.create();
+    const startDate = "2024-01-01";
+    const endDate = "2024-01-31";
+
+    const mockUsageData = [
```

Comment:
This looks like a case for some seeder with faker.

Reply:
Created and used `BillingUsageSeeder` in such cases